src/docs/development/add-to-app/android/add-flutter-screen.md

```diff
@@ -122,78 +48,79 @@ myButton.addOnClickListener(new OnClickListener() {
   @Override
   public void onClick(View v) {
     startActivity(
-      FlutterActivity.createBuilder().build(currentActivity)
+      FlutterActivity.createDefaultIntent(currentActivity)
     );
   }
 });
 ```
 
 The above example assumes that your Dart entrypoint is called `main()`, and your
-initial Flutter route is '/'. If you'd like to change either property, use the
-`IntentBuilder` to provide your desired properties:
+initial Flutter route is '/'. The Dart entrypoint cannot be changed via `Intent`,
+but the initial route can be changed via `Intent`. The following example
+demonstrates how to launch a `FlutterActivity` that initially renders a custom
+route in Flutter.
 
 ```java
 myButton.addOnClickListener(new OnClickListener() {
   @Override
   public void onClick(View v) {
     startActivity(
       FlutterActivity
-        .createBuilder()
-        .dartEntrypoint("mySpecialEntrypoint")
+        .withNewEngine()
         .initialRoute("/my_route")
         .build(currentActivity)
       );
   }
 });
 ```
 
-Replace `"mySpecialEntrypoint"` with the name of your desired Dart entrypoint,
-and replace `"/my_route"` with your desired initial route.
+Replace `"/my_route"` with your desired initial route.
+
+The use of the `withNewEngine()` factory method configures a `FlutterActivity`
+that internally create its own `FlutterEngine` instance. This comes with a 
+non-trivial initialization time. The alternative approach is to instruct
+`FlutterActivity` to use a pre-warmed, cached `FlutterEngine`, which minimizes
+Flutter's initialization time. That approach is discussed next.
 
-### Step 5: (Optional) Use a cached FlutterEngine
+### Step 3: (Optional) Use a cached FlutterEngine
 
 Every `FlutterActivity` creates its own `FlutterEngine` by default. Each
 `FlutterEngine` has a non-trivial "warm-up" time. This means that launching a
 standard `FlutterActivity` comes with a brief delay before your Flutter
-experience becomes visible. To avoid this delay, you can warm-up a
+experience becomes visible. To minimize this delay, you can warm-up a
 `FlutterEngine` before arriving at your `FlutterActivity` and then you can use
 your pre-warmed `FlutterEngine` instead.
 
 To pre-warm a `FlutterEngine`, find a reasonable location in your app to
-instantiate and hold a `FlutterEngine`. For a simplistic app, storing a
+instantiate a `FlutterEngine`. For a small app or a prototype, pre-warming a
```

Comment:
This makes it sound like it would be inappropriate to do so in a larger app where it might still be sensible to do if the flutter screen is the second screen and has a high probability of opening